src/main/java/Terminator.java

```diff
@@ -0,0 +1,648 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.StringJoiner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class Terminator {
+
+    /**
+     * Initialize the scanner to be used globally.
+     */
+    private static final Scanner scanObject = new Scanner(System.in);
+
+    /**
+     * Keywords for commands.
+     */
+    private static final String BYE_STRING = "BYE";
+    private static final String DONE_STRING = "DONE";
+    private static final String LIST_STRING = "LIST";
+    private static final String TODO_STRING = "TODO";
+    private static final String EVENT_STRING = "EVENT";
+    private static final String DELETE_STRING = "DELETE";
+    private static final String DEADLINE_STRING = "DEADLINE";
+
+    /**
+     * List of constants used to define who is speaking.
+     */
+    public static final String TERMINATOR_STRING = "[The Terminator]";
+    public static final String USER_STRING = "[User]";
+
+    /**
+     * List of constants used in extractNameDateTime
+     */
+    public static final String BY_KEYWORD = "/by";
+    public static final String AT_KEYWORD = "/at";
+    public static final String TODO_KEYWORD = "todo";
+    public static final String EVENT_KEYWORD = "event";
+    public static final String DEADLINE_KEYWORD = "deadline";
+
+    /**
+     * List of constants used in formatWithHeading.
+     */
+    public static final int TERMINATOR_FORMATTING = 0;
+    public static final int USER_FORMATTING = 1;
+
+    /**
+     * List of constants used in tokenizing user input.
+     */
+    public static final int KEYWORD_INDEX = 0;
+    public static final int TASK_NUMBER_INDEX = 1;
+
+    /**
+     * List of constants used to identify Event Type for createEvent
+     */
+    public static final String TODO_TYPE = "T";
+    public static final String EVENT_TYPE = "E";
+    public static final String NORMAL_TYPE = "N";
+    public static final String DEADLINE_TYPE = "D";
+
+    /**
+     * List of constants used to show index of taskName and dateTime from retrieval list
+     */
+    public static final int TASK_NAME_INDEX = 0;
+    public static final int DATE_TIME_INDEX = 1;
+    public static final int COMPLETION_INDEX = 2;
+
+    /**
+     * Global variable used to show if loop to get user input should continue running.
+     */
+    public static Boolean toContinue = true;
```

Comment:
Consider standardizing your use of boolean type to use either the Boolean object or boolean primitive so that your code is consistent. While `Boolean` and `boolean` essentially work the same (`Boolean` object provides more functionality under the hood), it is recommended that you stick to either one of them for consistency purposes.